src/main/java/bbm/brickbreaker/Controller.java

```diff
@@ -1,6 +1,74 @@
 package bbm.brickbreaker;
 
+import javax.swing.Timer;
+
+import java.awt.*;
+
+import static bbm.brickbreaker.Bounds.*;
+
 public class Controller {
-    // pass into frame width and height
-    // pass into brick width and height that is smaller than frame's
+    private final Ball ball;
+    private final Brick brick;
+    private final Panel panel;
+    private GameComponent view;
+
+    private int radius;
+
+    private Timer timer;
+
+    public Controller(Ball ball, Brick brick, int radius, Panel panel, GameComponent view) {
+        this.ball = ball;
+        this.brick = brick;
+        this.panel = panel;
+        this.radius = radius;
+        this.view = view;
+    }
+
+    public void play() {
+        timer = new Timer(1000 / 60, e -> {
+            double newX = ball.locationX();
+            double newY = ball.locationY();
+
+            ball.setPosition(newX, newY);
+            view.repaint();
+
+            Bounds hitDirection = NONE;
+
+            if (ball.hitsWall(radius)) {
+                if (newX - radius <= 0) {
+                    hitDirection = LEFT;
+                } else if (newX + radius >= ball.getWidth()) {
+                    hitDirection = RIGHT;
+                } else if (newY - radius <= 0) {
+                    hitDirection = TOP;
+                } else if (newY + radius >= ball.getHeight()) {
+                    timer.stop();
+                }
+
+                if (hitDirection != NONE) {
+                    ball.bounce(hitDirection);
+                }
+            }
+
+            checkPanelCollision();
+        });
+
+        timer.start();
+    }
+
+    private void breakBricks(int x, int y) {
+        if (brick.isBrick(x, y)) {
+            brick.hitBrick(x, y);
+            ball.bounce(Bounds.TOP);
+        }
+    }
```

Comment:
Get the position of the bricks here so you know where the bricks are for collision